CMakeLists.txt

```diff
@@ -1,28 +1,25 @@
 ########################################################################
 # Preamble
 ########################################################################
-
 cmake_minimum_required( VERSION 3.14 )
 project( ACEtk LANGUAGES CXX )
 
-
 ########################################################################
 # Project-wide setup
 ########################################################################
 
 set( CMAKE_CXX_STANDARD 17 )
 set( CMAKE_CXX_STANDARD_REQUIRED YES )
 
-option( ACEtk_unit_tests
-    "Compile the ACEtk unit tests and integrate with ctest" ON
-    )
+option(ACEtk_unit_tests "Build the ACEtk unit tests" OFF)
+option( ACEtk.python "Build ACEtk python bindings" ON )
 option( strict_compile
     "Treat all warnings as errors." ON
     )
 
 # Compile flags
-set( common_flags "-Wall" "-Wextra" "-Wpedantic" )
-set( strict_flags "-Werror" )
+set( common_flags "-Wall" "-Wextra" "-Wpedantic" ) # i'm not sure why we manage flags this way...
+set( strict_flags "-Werror" )                      # improving this will need to happen in the future for spack
```

Comment:
I took this from our previous build system based on Austin's metaconfigure.  That had a much more robust flag management, with flags set by compiler, etc.  I slimmed it down for now, but we may need more of the old functionality back at some point.

Always happy to see suggestions for improvements :)

Note that these options like strict_compile aren't even hooked up to the build step -- that's a sloppy work-in-progress issue that I haven't gone back to fix.  So for sure... work to be done!

Reply:
My suggestion would be to use the defaults provided by cmake to begin. We can add custom build types in a `CustomBuildTypes.cmake` file where we capture the extra stuff we have here. The reasoning is that we don't want to break from convention if possible. Especially, once we move towards providing spack recipes because spack will take care of adding platform specific flags like `march and mtune`.

Reply:
BTW not offended at all by formatting preferences - what are they?

Reply:
Last thing - what is devtools?